brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -107,7 +107,7 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
         final RequestHeadersBuilder newHeaders = req.headers().toBuilder();
         final HttpClientRequest request = ClientRequestContextAdapter.asHttpClientRequest(ctx, newHeaders);
         final Span span = handler.handleSend(request);
-        req = HttpRequest.of(req, newHeaders.build());
+        req = req.withHeaders(newHeaders.build());
```

Comment:
We can do it in separate PR but noticed that it'd be reasonable to add `withHeaders(RequestHeadersBuilder h) { withHeaders(h.build()); }` - inspired by protobuf which generally accepts both built and builders in its methods I think

Reply:
That's a good idea. Added.